File: word-pattern/word-pattern.cpp
class Solution {
public:
    bool wordPattern(string pattern, string s) {
        vector<string> st;
        
        string k;
        int i,n=s.size();
        
        for(i=0;i<n;i++)
        {
            if(s[i]-0==32)
            {
                st.push_back(k);
                
                k.clear();
            }
            else
            {
                k.push_back(s[i]);
            }
        }
        st.push_back(k);
        k.clear();
        
        unordered_map<char,string> mp;
        unordered_map<string,char> mp2;
        n=pattern.size();
        if(n!=st.size())
            return false;
        for(i=0;i<n;i++)
        {
            if(mp.find(pattern[i])==mp.end())
            {
                mp.insert({pattern[i],st[i]});
            }
            else
            {
                auto it=mp.find(pattern[i]);
                if(it->second!=st[i])
                    return false;
            }
            if(mp2.find(st[i])==mp2.end())
            {
                mp2.insert({st[i],pattern[i]});
            }
            else
            {
                auto it=mp2.find(st[i]);
                if(it->second!=pattern[i])
                    return false;
            }
        }
        
        return true;
    }
};

File: word-pattern/word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        str1 = "";
        str2 = "";
        occur = 1;
        mp = {'':0}
        for ch in pattern:
            if ch not in mp:
                mp[ch] = occur;
                str1+=(str)(occur);
                occur+=1;
            else:
                str1+=str(mp[ch]);
                
        print(str1)
        arr = s.split(" ");
        mp1 = {'':0}
        oc = 1;
        for ch in arr:
            if ch not in mp1:
                mp1[ch] = oc;
                str2+=(str)(oc);
                oc+=1;
            else:
                str2+=str(mp1[ch]);
        print(str2)
        return str1 == str2;
            
        